src/ParallelAlgorithms/EventsAndTriggers/Tags.hpp

```diff
@@ -80,14 +84,18 @@ struct ObservationValue {
 namespace Tags {
 /// \ingroup EventsAndTriggersGroup
 /// Contains the events and triggers
+template <Triggers::WhenToCheck WhenToCheck>
 struct EventsAndTriggers : db::SimpleTag {
   using type = ::EventsAndTriggers;
```

Comment:
From what I recall, the original plan was to template the `Event` class (and possibly `Trigger`) as well, since most events won't work if run `AtSteps`.  Was that idea abandoned?

Reply:
No; that is what I plan to do in the next PR.  I just thought it would be easier to break things up into separate PRs, as this one will be the one that requires changes to the input files, and I wanted to get the enum finalized before templating the Events on it.

Reply:
OK, that's fine.